flocker/node/agents/test/test_blockdevice.py

```diff
@@ -2310,7 +2310,12 @@ def test_profile_respected(self):
         mandatory profiles.
         """
         for profile in (c.value for c in MandatoryProfiles.iterconstants()):
-            self.api.create_volume_with_profile(profile)
+            size = GiB(1).to_Byte().value
+            dataset_id = uuid4()
+            self.addCleanup(detach_destroy_volumes, self.api)
+            self.api.create_volume_with_profile(dataset_id=dataset_id,
+                                                size=size,
+                                                profile_name=profile)
 
```

Comment:
It would be good if the standard interface tests checked that something created as GOLD is returned as GOLD; that after all is the point of a backend saying they implement the interface. Possibly the test class should be constructed with constants indicating size to use per profile so you don't have to hard code a value that works for all profiles on all backends.